### PR TITLE
Feat/expose prom metrics

### PR DESCRIPTION
**Pull request by LukasDeco**
Adds a GET http endpoint with express(should we use something else? Fastify?) that exposes prometheus metrics. 
Added a logger class for incrementing error and warn metrics. So far we just have errors it seems to log and increment.

Notes:
Had to used a forked version of prom-client. I'll open a PR with my fork and then when they merge it in(hopefully) we can go back to using the real library.

There are tons of formatting changes from my prettier... maybe if we all use same formatter we can agree on print width and tabs and all that.

### CODE REVIEW

**Comment by R-K-H** [comment]
So this looks good to me, at first I was confused with the changes but realized linting and then the fact that the metrics are intended to exist here. I'm assuming that prometheus is running in another instance or is that TBD?

Also I think @decartesol should review before merge, but reviewed it and it seems good to me.

**Review by R-K-H — APPROVED** [review]

**Comment by LukasDeco** [comment]
> So this looks good to me, at first I was confused with the changes but realized linting and then the fact that the metrics are intended to exist here. I'm assuming that prometheus is running in another instance or is that TBD?
> 
> Also I think @decartesol should review before merge, but reviewed it and it seems good to me.

Yeah prometheus does run on a different instance generally. I'm trying to see if prometheus and grafana can be run in the same container so we just add one more container instance to the railway project for backend.

**Comment by decartesol** [comment]
I really appreciate the clean API of that logger.

I'm adding a few small changes then will merge

**Review by decartesol — APPROVED** [review]

**Pull request merged**
